Documentation/DesignGuidelines.md

```diff
@@ -6,28 +6,459 @@ Guidelines](http://blogs.msdn.com/b/rxteam/archive/2010/10/28/rx-design-guidelin
+the closure passed to the [`startWithSignal`][startWithSignal] method.
+
+#### Prefer managing lifetime with operators over explicit disposal
+
+Although the [disposable][Disposables] returned from [`start`][start] makes
+canceling a [signal producer][Signal Producers] really easy, explicit use of
+disposables can quickly lead to a rat's nest of resource management and cleanup
+code.
+
+There are almost always higher-level [operators][] that can be used instead of manual
+disposal:
+
+ * [`take`][take] can be used to automatically terminate a stream once a certain
+   number of values have been received.
+ * [`takeUntil`][takeUntil] can be used to automatically terminate
+   a [signal][Signals] or producer when an event occurs (for example, when
+   a “Cancel” button is pressed in the UI).
+ * [Properties][] and the `<~` operator can be used to “bind” the result of
+   a signal or producer, until termination or until the property is deallocated.
+   This can replace a manual observation that sets a value somewhere.
 
 ## Implementing new operators
 
+RAC provides a long list of built-in [operators][] that should cover most use
+cases; however, RAC is not a closed system. It's entirely valid to implement
+additional operators for specialized uses, or for consideration in ReactiveCocoa
+itself.
 
-[Framework Overview]: FrameworkOverview.md
+Implementing a new operator requires a careful attention to detail and a focus
+on simplicity, to avoid introducing bugs into the calling code.
+
+These guidelines cover some of the common pitfalls and help preserve the
+expected API contracts. It may also help to look at the implementations of
+existing `Signal` and `SignalProducer` operators for reference points.
+
+#### Prefer writing operators that apply to both signals and producers
+
+Since any [signal operator can apply to signal
+producers](#signal-operators-can-be-lifted-to-apply-to-signal-producers),
+writing custom operators in terms of [`Signal`][Signals] means that
+[`SignalProducer`][Signal Producers] will get it “for free.”
+
+Even if the caller only needs to apply the new operator to signal producers at
+first, this generality can save time and effort in the future.
 
+Of course, some capabilities _require_ producers (for example, any retrying or
+repeating), so it may not always be possible to write a signal-based version
+instead.
+
+#### Compose existing operators when possible
+
+Considerable thought has been put into the operators provided by RAC, and they
+have been validated through automated tests and through their real world use in
+other projects. An operator that has been written from scratch may not be as
+robust, or might not handle a special case that the built-in operators are aware
+of.
+
+To minimize duplication and possible bugs, use the provided operators as much as
+possible in a custom operator implementation. Generally, there should be very
+little code written from scratch.
+
+#### Forward error and interruption events
```

Comment:
What about completion events? 
Is this section here to emphasise that error and interruption should be forwarded __asap__, or as a reminder that they need to be forwarded at all?